Sources/NIOFileSystem/FileHandleProtocol.swift

```diff
@@ -152,6 +143,26 @@ public protocol FileHandleProtocol {
     ///
     /// After closing the handle calls to other functions will throw an appropriate error.
     func close() async throws
+
+    /// Sets the file's last access and last data modification times to the given values.
+    ///
+    /// If **either** time is `nil`, the current value will not be changed.
+    /// If **both** times are `nil`, then both times will be set to the current time.
```

Comment:
IMO ["clamp/update the value to the closest valid value"](https://github.com/apple/swift-nio/pull/2735/files/8222158a110c16d705f767fd3e53caeca910aa56..d3cf9efe57bff7a541f0edf24dcda48dd0d4962a#r1631408774) is good/fine, but the API design still bothers me.

I would have preferred these 2 lines to have been obvious in the API design.
Perhaps using an (structified) enum as the single parameter that the functions accepts. Or anything better. Maybe splitting up the responsibility into 2/3 different functions.

Reply:
I think it's clear from `setTimes(lastAccess: t)` that `lastDataModification` isn't modified. 

It's fair to say though that setting both to `nil` _isn't_ an obvious API: `setTimes(lastAccess: nil, lastDataModification: nil)` but that's why we also have the `touch()` wrapper which many developers _are_ familiar with.

Reply:
If we had separate methods for each, we'd unavoidably have to make two separate sys calls, which is expensive. We'd also not be setting the times to the exact same value, as the current time will have changed if we use the `UTIME_NOW` `timespec`.

However, I've added two new convenience methods that default one of the parameters to `nil` to have nicer API: `setLastAccessTime(to:)` and `setLastDataModificationTime(to:)`.